### PR TITLE
Cloudwatch: Refactoring - move describe log groups call to CallResourceHandler

### PR DESCRIPTION
**Pull request by sunker**
**What this PR does / why we need it**:

This PR moves the describeLogGroups api call from the QueryDataHandler to the CallResourceHandler. 

The `describeLogGroups` call is used for discovery data, i.e data that is used for composing the query in the editor. By convention, these type of queries should use the CallResourceHandler and not the QueryDataHandler. 

While doing this refactoring, I've been trying to fix as many Betterer issues as possible. 

The `handleDescribeAllLogGroups` will be moved to the resource api in an upcoming PR.  

**Which issue(s) this PR fixes**:

Fixes #https://github.com/grafana/grafana/issues/55243
Fixes #https://github.com/grafana/cloud-data-sources/issues/34

Part of https://github.com/grafana/grafana/issues/55548
**Special notes for your reviewer**:
Sorry for the large amount of changed files, but I don't think this PR could be split into multiple ones.

### CODE REVIEW

File: .betterer.results [hunk 1]
```diff
@@ -56,7 +56,7 @@ exports[`no enzyme tests`] = {
     "public/app/features/dimensions/editors/ThresholdsEditor/ThresholdsEditor.test.tsx:145048794": [
       [0, 17, 13, "RegExp match", "2409514259"]
     ],
-    "public/app/plugins/datasource/cloudwatch/components/ConfigEditor.test.tsx:2210656642": [
+    "public/app/plugins/datasource/cloudwatch/components/ConfigEditor.test.tsx:2983010995": [
```
Comment (joshhunt) [inline .betterer.results:59]: ✅

File: public/app/plugins/datasource/cloudwatch/api.test.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  it('replaces region correctly in the query', async () => {
+    const { api, resourceRequestMock } = setupMockedAPI();
+    await api.describeLogGroups({ region: 'default' });
+    expect(resourceRequestMock.mock.calls[0][1].region).toBe('us-west-1');
+
+    await api.describeLogGroups({ region: 'eu-east' });
+    expect(resourceRequestMock.mock.calls[1][1].region).toBe('eu-east');
+  });
+
+  it('should return log groups as an array of strings', async () => {
```
Comment (iwysiu) [inline public/app/plugins/datasource/cloudwatch/api.test.ts:13]: This name technically isn't true anymore

**Review by iwysiu — APPROVED** [review]
overall nice! some comments.

File: pkg/tsdb/cloudwatch/log_actions.go [hunk 3]
```diff
@@ -203,40 +201,6 @@ func (e *cloudWatchExecutor) handleGetLogEvents(ctx context.Context, logsClient
...
-		logGroupNames = append(logGroupNames, logGroup.LogGroupName)
-	}
-
-	groupNamesField := data.NewField("logGroupName", nil, logGroupNames)
-	frame := data.NewFrame("logGroups", groupNamesField)
-
-	return frame, nil
-}
-
 func (e *cloudWatchExecutor) handleDescribeAllLogGroups(ctx context.Context, logsClient cloudwatchlogsiface.CloudWatchLogsAPI, parameters LogQueryJson) (*data.Frame, error) {
```
[the remark this answers is not in the document]
Comment (sunker) [reply pkg/tsdb/cloudwatch/log_actions.go:204]: Yes will handle it in a separate PR. See comment in description. :+1:

**Pull request merged**